static/panes/compiler.js

```diff
@@ -1184,6 +1184,8 @@ Compiler.prototype.setAssembly = function (result, filteredCount) {
     this.assembly = asm;
     if (!this.outputEditor || !this.outputEditor.getModel()) return;
     var editorModel = this.outputEditor.getModel();
+    if (result.languageId !== 'undefined')
```

Comment:
I love the idea of having the ID be part of the `result`.  I'd prefer using an explicit `null` or not having the `languageId` at all, in the cases where ther eisn't instead of a magic string `'undefined'`